addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -8362,6 +8382,70 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
             });
         }
     },
+    /**
+     * @override
+     */
+    notify(name, data) {
+        if (name == "bgColorChanged") {
+            if (this.$target[0].querySelector(".o_we_shape")) {
+                this._handlePreviewState(data.preview, () => {
+                    const shape = this._getShapeData().shape;
+                    return { colors: this._getImplicitColors(shape) };
+                });
+            }
+            this._updateShapeForNeighbor(data.preview, this.$target[0].nextElementSibling, true);
+            this._updateShapeForNeighbor(
+                data.preview,
+                this.$target[0].previousElementSibling,
+                false
+            );
+        } else {
+            this._super(...arguments);
+        }
+    },
+    /**
+     * Set the background color of the connection shape of the previous and next snippet.
+     * @param {boolean} previewMode
+     * @param {Element} snippetEl the (next/previous)ElementSibling of the current snippet.
+     * @param {boolean} shouldBeFlip Indicates if the connection shape has to be flip to be next
+     * to the current snippet (true for the next snippet and false for the previous).
+     */
+    _updateShapeForNeighbor(previewMode, snippetEl, shouldBeFlip) {
+        if (!snippetEl || !snippetEl.querySelector(".o_we_shape")) {
+            return;
+        }
+        const shapeData = this._getShapeData(snippetEl);
+        const flipCheck = shapeData.flip.includes("y") === shouldBeFlip;
+        if (!(shapeData.shape.includes("web_editor/Connections") && flipCheck)) {
+            return;
+        }
+        const shapeEl = snippetEl.querySelector(".o_we_shape");
+        let shapeBackgroundImage = window
+            .getComputedStyle(shapeEl)
+            .getPropertyValue("background-image");
+        if (previewMode === true) {
+            snippetEl.dataset.originalBackgroundImage = shapeBackgroundImage;
+        } else if (previewMode === false) {
+            snippetEl.dataset.rerender = true;
+            delete snippetEl.dataset.originalBackgroundImage;
+        } else if (previewMode === "reset") {
+            shapeEl.style.backgroundImage = snippetEl.dataset.originalBackgroundImage;
+            delete snippetEl.dataset.originalBackgroundImage;
+            return;
+        }
```

Comment:
Normally saving the original state is not necessary: after preview we re-apply the current option (but to confirm, sometimes it's more complicated than that).